contracts/FTokenStrat.sol

```diff
@@ -0,0 +1,212 @@
+        IERC20(_token).safeTransfer(_to, IERC20(_token).balanceOf(address(this)));
+    }
+
+    // Bypasses withdrawal cap. Should be used with care. Can cause slippage with large amounts.
+    function withdrawShares(uint shares) public onlyStrategist {
+        fToken.withdraw(shares);
+    }
+
+    // Bypasses withdrawal cap. Should be used with care. Can cause slippage with large amounts.
+    function withdrawUnderlying(uint amount) public onlyStrategist {
+        fToken.withdraw(sharesForAmount(amount));
+    }
+
+    // Bypasses withdrawal cap. Should be used with care. Can cause slippage with large amounts.
+    function withdrawAll() public onlyStrategist {
+        fToken.withdrawAll();
+    }
+
+    function depositUnderlying(uint amount) public onlyStrategist {
+        fToken.deposit(amount);
+    }
+
+    function depositAll() public onlyStrategist {
+        fToken.deposit(underlying.balanceOf(address(this)));
+    }
+
+    // set buffer to -1 to pause deposits. 0 to remove buffer.
+    function setBuffer(uint _buffer) public onlyStrategist {
+        buffer = _buffer;
+    }
+
+    // set to -1 for no cap
+    function setWithdrawalCap(uint underlyingCap) public onlyStrategist {
+        require(underlyingCap >= minWithdrawalCap);
+        withdrawalCap = underlyingCap;
+    }
+
+    function sharesForAmount(uint amount) internal view returns (uint) {
+        return amount.mul(fToken.totalSupply()).div(fToken.underlyingBalanceInVault());
+    }
+
+    function setStrategist(address _newStrategist) public onlyOwner {
+        strategist = _newStrategist;
+    }
+
+    function changeTimelock(Timelock _newTimelock) public onlyTimelock {
+        timelock = Timelock(_newTimelock);
+    }
+
+    function changeRewardPool(IRewardPool _newRewardPool ) public onlyOwner {
+        require(buffer == uint(-1), "DEPOSITS MUST BE PAUSED");
+
+        uint transitBalance = rewardpool.balanceOf(address(this));
+        rewardpool.withdraw(transitBalance);
+        rewardpool = _newRewardPool;
+        fToken.approve(address(rewardpool), uint(-1));
+
+        rewardtoken = IERC20Detailed(rewardpool.rewardToken());
+        rewardtoken.safeApprove(address(rewardpool), uint(-1));
+        _newRewardPool.stake(transitBalance);
+    }
+
+    function harvestRewardToken(uint outMin, address[] calldata path, uint deadline) public onlyStrategist returns (uint) {
+        uint received = 0;
+        uint rewardTokenBalance = rewardtoken.balanceOf(address(this));
+        if(rewardTokenBalance > uint(0)){
+          IERC20 to = vault.underlying();
+          rewardtoken.approve(address(router), rewardTokenBalance);
+          uint[] memory amounts = router.swapExactTokensForTokens(rewardTokenBalance, outMin, path, address(this), deadline);
+          received = amounts[amounts.length -1];
+
+          to.approve(address(vault), received);
+          underlying.safeTransfer(address(vault), received);
```

Comment:
Rewards should not be sent to the vault. They should stay in the strategy where they will be re-invested automatically during the next call to `invest()`

```suggestion
          rewardtoken.approve(address(router), rewardTokenBalance);
          uint[] memory amounts = router.swapExactTokensForTokens(rewardTokenBalance, outMin, path, address(this), deadline);
          received = amounts[amounts.length -1];
```